stored_requests/fetcher.go

```diff
@@ -52,16 +52,23 @@ type fetcherWithCache struct {
 
 func (f *fetcherWithCache) FetchRequests(ctx context.Context, ids []string) (data map[string]json.RawMessage, errs []error) {
 	data = f.cache.GetRequests(ctx, ids)
-	if len(data) >= len(ids) {
+
```

Comment:
Wouldn't it be easier to check leftoverIds below, and then if == 0 "return data, nil", rather than doing this extra loop? That still gets us out before trying to look up an empty list.

Reply:
Actually on looking into it... we don't even need that. The issue is fixed just by changing the way I generate the IDs we send to the `Fetcher`.

It also surfaced another extra check I could delete. Since we don't know whether the PBS host is running with a cache, the `Fetcher` needs to be smart enough to return quickly on empty lists in the first place. Given that, it means there's also no real value in doing check to see if we have an empty list before making the function call.

Stuff is much simplified now.